src/Plugin/EntityReferenceSelection/OgSelection.php

```diff
@@ -60,7 +64,14 @@ protected function buildEntityQuery($match = NULL, $match_operator = 'CONTAINS')
     // the default selection handler of the entity, which the field reference
     // to, and add another logic to the query object i.e. check if the entities
     // bundle defined as group.
-    $query = $this->getSelectionHandler()->buildEntityQuery();
+    $selection_handler = $this->getSelectionHandler();
+
+    if (empty($selection_handler)) {
+      return $query = parent::buildEntityQuery($match, $match_operator);
```

Comment:
> That will still leave us with a fatal as well.

Indeed, which I understand the scenario correctly, I think is the right outcome. If the field you expected to have is badly configured - you'd want to know about it loudly. Or am I missing something? :)